### ISSUE TITLE
Find out if Cache-Control: public is necessary for proxies to cache content 

### ISSUE THREAD

**Issue by GoogleCodeExporter** [pagespeed/page-speed#9]
```
'm wondering if any caches need this directive to tell them to
cache these resources. The HTTP RFC indicates that CC: public can be
used to indicate that a "shared cache" can return a locally cached
response when the request contains an "Authorization" header:

http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.8

And the docs for CC: public are quite vague:

http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.9.1

"Indicates that the response MAY be cached by any cache, even if it
would normally be non-cacheable or cacheable only within a non- shared
cache."

"even if it would normally be non-cacheable" seems bad - does that
mean that CC: public implies that the resource has an infinite
"freshness lifetime"?

This section implies that anything cacheable by the client is also
cacheable by shared proxies "unless specifically constrained by a
cache-control directive":
http://www.w3.org/Protocols/rfc2616/rfc2616-sec13.html#sec13.4

In short, everything in the RFC implies that a plain old response with
a future Expires and no Cache-Control header would be cacheable by an
HTTP proxy, and that CC: public is not needed.

squid and possibly other proxies do not require this header in order to 
cache content. Before we warn when CC: public is missing, we need to find 
out if it improves proxy cache hitrates first.
```

Original issue reported on code.google.com by `bmcqu...@google.com` on 3 Jun 2009 at 11:10
